runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -2789,27 +2789,33 @@ class MM_ObjectAccessBarrierAPI
 	/**
 	 * Check to see if the object is marked
 	 *
-	 * @NOTE only for WRT as this check the mark map for segregated heaps
+	 * @NOTE only for SATB Concurrent Mark
 	 *
 	 * @param object to check
 	 * @return true if the object was marked, false otherwise
 	 */
 	VMINLINE bool
 	isMarked(J9VMThread *vmThread, j9object_t object)
 	{
-		UDATA heapMapBits = 0;
 		J9JavaVM *const javaVM = vmThread->javaVM;
-		UDATA heapDelta = (UDATA)object - javaVM->realtimeHeapMapBasePageRounded;
-		UDATA slotIndex = heapDelta >> J9VMGC_SIZECLASSES_LOG_SMALLEST;
- 		UDATA bitIndex = slotIndex & J9_GC_MARK_MAP_UDATA_MASK;
-		UDATA bitMask = ((UDATA)1) << bitIndex;
-		slotIndex = slotIndex >> J9_GC_MARK_MAP_LOG_SIZEOF_UDATA;
 
-		UDATA *heapMapBitsBase = javaVM->realtimeHeapMapBits;
-		heapMapBits = heapMapBitsBase[slotIndex];
-		heapMapBits &= bitMask;
+		if (J9_GC_POLICY_METRONOME == javaVM->gcPolicy) {
+			UDATA heapMapBits = 0;
+			UDATA heapDelta = (UDATA)object - javaVM->realtimeHeapMapBasePageRounded;
+			UDATA slotIndex = heapDelta >> J9VMGC_SIZECLASSES_LOG_SMALLEST;
+			UDATA bitIndex = slotIndex & J9_GC_MARK_MAP_UDATA_MASK;
+			UDATA bitMask = ((UDATA)1) << bitIndex;
+			slotIndex = slotIndex >> J9_GC_MARK_MAP_LOG_SIZEOF_UDATA;
 
-		return (0 != heapMapBits);
+			UDATA *heapMapBitsBase = javaVM->realtimeHeapMapBits;
+			heapMapBits = heapMapBitsBase[slotIndex];
+			heapMapBits &= bitMask;
+
+			return 0 != heapMapBits;
+		} else {
+			/* TODO: Inline for SATB Standard Configuration */
+			return javaVM->memoryManagerFunctions->j9gc_ext_is_marked(vmThread->javaVM, object);
```

Comment:
One compiler is (expectedly) complaining about this UDATA being implicitly downcasted to bool.
I guess, one way to fix it is to change the return type of  j9gc_ext_is_marked to BOOLEAN? (I prefer a solution without explicit casting, if possible)

Reply:
`BOOLEAN` is not significantly different from `UDATA`; `omrcomp.h` says (for some platforms):
```
#define BOOLEAN UDATA
```
The proper way to map scalar types to `bool` is by comparing with zero:
```
	return 0 != javaVM->memoryManagerFunctions->j9gc_ext_is_marked(vmThread->javaVM, object);
```

Reply:
yeah, I was suspecting it could be identical. 
ok, your suggestion is good. although having j9gc_ext_is_marked return BOOLEAN would be cleaner, too (in which case we compare the return with TRUE/FALSE, instead)

Reply:
I've changed it based on @keithc-ca suggestion, it's the least amount of changes required and it's consistent with what is being done a few lines up.  But I think that BOOLEAN would have been cleaner.  This code will be revisited to inline the check, so perhaps we can consider changing it then.